Source/WebCore/rendering/svg/SVGResourcesCache.cpp

```diff
@@ -79,6 +79,16 @@ SVGResources* SVGResourcesCache::cachedResourcesForRenderer(const RenderElement&
     return resourcesCacheFromRenderer(renderer).m_cache.get(&renderer);
 }
 
+static bool hasPaintResourceRequiringRemovalOnClientLayoutChange(RenderSVGResource* resource)
+{
+    return resource && resource->resourceType() == PatternResourceType;
+}
+
+static bool hasResourcesRequiringRemovalOnClientLayoutChange(SVGResources& resources)
+{
+    return resources.clipper() || resources.masker() || resources.filter() || hasPaintResourceRequiringRemovalOnClientLayoutChange(resources.fill()) || hasPaintResourceRequiringRemovalOnClientLayoutChange(resources.stroke());
```

Comment:
What should happen if SVGResources has a linked resource and this linked resource is a gradient?

`<linearGradient id="gradient2" href="#gradient1"/>`

Reply:
I'm adding a couple of tests for invalidation with linked resources in https://bugs.webkit.org/show_bug.cgi?id=242497.